### PR TITLE
Try more universal rich text controls

### LINKED ISSUE #2086: [Icon with text] Change the Heading "text" setting to "inline_richtext" setting
**Describe the enhancement you'd like**
Change the Heading `text` setting for `inline_richtext` to let merchants include links, bold and italic styling on this text block. This will allow merchants to link to other areas of their online stores and align with the block intention.

**Describe alternatives you've considered**
We could have used the `richtext` section as well but since the text setting doesn't support line breaks and that line breaks are not optimize for this use case, `inline_richtext` is a better improvement.

**Additional context/screenshots**
Maybe a screenshot or design?

- [Before](https://screenshot.click/01-41-oqj3g-tb7mi.png)
- [After](https://screenshot.click/01-47-9nsr6-dl037.png)

### PR DESCRIPTION
**Pull request by kjellr**
### PR Summary: 

This PR swaps `text` input settings out for `inline_richtext` controls in the following places: 

- ~Collapsible content > Heading~
- ~Collapsible content > Row name~
- Collection List > Heading
- Contact Form > Heading
- Featured Blog > Heading
- Featured Collection > Heading
- Featured Product > Text Block > Heading
- Featured Product > Icon with Text > Headings 1-3
- Featured Product > Collapsible Tab > Heading
- Footer > Link List Heading
- Footer > Text Block > Heading
- Footer > Newsletter Heading
- Image Banner > Heading
- Image Banner > Text
- Image with Text > Heading
- Main List Collections > Heading
- Main Product > Text Block > Heading
- Main Product > Icon with Text > Headings 1-3
- ~Main Product > Collapsible Tab > Heading~
- Multicolumn > Heading
- Multicolumn > Column > Title
- Multimedia Collage > Heading
- Newsletter > Heading
- Product Recommendations > Heading
- Slideshow > Heading
- Slideshow > Subheading
- Video > Heading

### Screenshot

<details>
  <summary>🎥 Demo video</summary>
  
  https://user-images.githubusercontent.com/1202812/200848984-3fcd41d6-8411-4e52-b9d6-5d2b8d4403a5.mp4
</details>

### Why are these changes introduced?

We currently don't universally allow merchants to perform simple inline text formatting (bold, italic, etc.). This is only turned on for a seemingly-random subset of different Sections and Blocks. This PR is an attempt to make those controls more universal. 

Fixes #2086 (and also attempts to fix that problem for other similar situations). 

### What approach did you take?

In general, I manually added various sections to my store and swapped out `text` inputs for `richtext` or `inline_richtext` whenever there wasn't a clear reason to avoid doing so. 

### Visual impact on existing themes

Based on my testing, I'm pretty sure this should have no impact — it only adds functionality.

### Demo links

[Demo Store](https://os2-demo.myshopify.com/admin/themes/136848474134/editor)

### Decision Log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |Do not allow `inline_richtext` in collapsible row headings.|Add `inline_richtext` in collapsible row headings. |We decided not to enable this because [it interferes](https://github.com/Shopify/dawn/pull/2085#discussion_r1039830528) with the clickable expanding/collapsing hover states.|Less flexibility in customizing the text within collapsible row headings.|
| 2 |Do not allow `inline_richtext` in collapsible row headings within `main-product.liquid`.|Allow it there.|Although collapsible rows here do not have underlined hover states like they do in the Collapsible Row block, we decided that for consistency with decision No. 1, we should leave `inline_richtext` functionality out of here too.|Less flexibility in customizing the text within collapsible row headings.|
|3|Do not update the existing `richtext` field for Rich Text Section's Heading block to use `inline_richtext` instead.|Update it|Updating this field would introduce the potential of losing merchant-entered data on theme upgrade. We would like to explore possible solutions to this this in a followup PR.|Unlike most other headings fields, merchants can enter multiple paragraphs into this field. These will continue to be stripped out on render.|
|4|We are going to punt the decision of whether to commit to using `richtext` in the Image Banner and SlideShow Slide description fields.|Enable richtext for these areas.|We'd considered doing this to align on using `richtext` here to align with the Image with Text Section's handling of this area, but [ran into bugs](https://github.com/Shopify/dawn/pull/2085#discussion_r1039844658) that would require more intense work. Upgrading to just `inline_richtext` is straightforward and introduces none of these issues, so we'll take this as a first step.|Merchants cannot use multi-paragraph text in the description fields for Slideshow and Image Banner.|

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [x] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by kjellr** [comment]
One thing to look out for in reviews: 

- [ ] Since some of these did not allow for inline links before, we need to make sure that they all include appropriate styles for any links that might be added in.

**Comment by elyseviotto** [comment]
Just catching up on this topic! 
I found interesting context in Slack about RTEs VS text / textarea: see https://shopify.slack.com/archives/CB0JT58RE/p1665156099604899

File: sections/collapsible-content.liquid
Comment (kjellr) [reply sections/collapsible-content.liquid:289]: Thanks, I missed that one. Added in d4e822b. 🎉

**Comment by kjellr** [comment]
> Since some of these did not allow for inline links before, we need to make sure that they all include appropriate styles for any links that might be added in.

This should be fixed as of a9531cb. I'd love a dev review of that commit specifically though (@tyleralsbury would you mind giving this a look, or pointing someone else my way?).

In this commit, I added a new `component-inline-rte.css` file to handle the link styling when we use the Inline RTE component. The normal `.rte` styles seemed overkill, and also [impose a link color](https://github.com/Shopify/dawn/blob/287add2dca7c64ea58f64daa52f426fceae530f5/assets/component-rte.css#L48), which felt wrong in this case. Some of the headings here [use a different color](https://screenshot.click/09-46-fmuu2-mnzxf.png), and we don't necessarily want to modify that color for links. I pulled the new file in via the header of each of the files I touched, but I'm not sure if that's actually necessary?

File: templates/index.json
Comment (kjellr) [inline templates/index.json:56]: One question I have: Now that this setting uses `inline_richtext` instead of `richtext`, these need to be removed to pass tests. Is this going to cause an issue with existing stores that might have stored content that includes `<p>` tags? 🤔

Comment (melissaperreault) [reply templates/index.json:56]: I am also wondering about the scenario merchants did add line breaks in that setting value `<br>` what will be their experience in how we map it out for a success theme update flow?

I mean my understanding it would make all <br> value invalid but would ignore them and do this:

>
    <p>This is a <br>
    sentence <br>
    written in <br>
    multiple lines</p>


to this :point_down:

>
    This is a sentence written in multiple lines


The only question I have is when I interact with the `inline_richtext` and try to add `<br>` manually and hit save, I end up in an error loop. I wonder if that’ll be the case for merchants.

Comment (melissaperreault) [reply templates/index.json:56]: We are all good finally, we can ignore 👼 the above ☝️

File: sections/collapsible-content.liquid
Comment (ludoboludo) [reply sections/collapsible-content.liquid]: Maybe that's something we can add to the PR description in a decision log table ? 

Here is the usual formatting we use, if you want to copy it over: 
| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 |   |   |   |   |

File: templates/index.json
Comment (kjellr) [reply templates/index.json:56]: Ah ok, got it!

File: assets/base.css
Comment (kjellr) [reply assets/base.css:734]: Yeah, I'm all for a new name. I think `richtext-container` is a little confusing, since we already have separate `rte` link styles. Would it be too weird to name it after its function? So something like `curentcolor-links`?

File: assets/base.css
Comment (kjellr) [reply assets/base.css:762]: Yeah, looks like we didn't need `.header-inline-links a` or `.banner:not(.slideshow) .header-inline-links a`. The other rules are written to keep the single-color links + hovers when there's a transparent background for Slideshow Slides and Image Banners.

Desktop

Before|After
---|---
image-banner-no-styles|image-banner-styles

Mobile

Before|After
---|---
mobile-image-banner-no-styles|mobile-image-banner-styles

File: assets/base.css
Comment (kjellr) [reply assets/base.css:734]: Ah no, I was just suggesting we keep the current solution exactly as it is, but call it `rte--inline`. So essentially the same as your second bullet here (but with a slightly different name). I'm cool with `inline-richtext` for now, so I'll update the PR to use that. 👍

File: sections/rich-text.liquid
Comment (melissaperreault) [reply sections/rich-text.liquid:168]: @kjellr for the same reason Ludo shared above I would be suggesting to go for `richtext` for image banner to keeping consistency among similar element.

Comment (kjellr) [reply sections/rich-text.liquid:168]: @melissaperreault and I chatted offline, and we want to clarify the situation here. 

We _are_ ok with losing inline formatting . So for example, if the merchant had `<br />` tags in there, and we're losing them upon upgrade, that's acceptable if we give them a heads up about it. 

But @ludoboludo, it sounds like [from your description up here](https://github.com/Shopify/dawn/pull/2085#discussion_r1040187842), we're losing the merchant's text entirely, is that correct? 

If that's the case, that's not ok. We don't want to have content loss on upgrade. If this is the case (and we can't fix it with something like [the transformation suggestion](https://github.com/Shopify/shopify/blob/main/components/online_store/app/models/theme/setting_updater.rb#L9)), then we should keep this field as is for now.

Comment (kjellr) [reply sections/rich-text.liquid]: Ok, thanks for the clarification. @melissaperreault and I don't want to lose any content, but we also don't want to hold up this PR. 

So for now, I've removed this `richtext` > `inline_richtext` change from the PR (https://github.com/Shopify/dawn/pull/2085/commits/610e6c42027ef3ef7c558c224ede7777a451e3f0). I did leave the `inline-richtext` class on the Rich Text Heading field though, so that the hover color for that matches the hover state for other headings (https://github.com/Shopify/dawn/pull/2085/commits/2659ebade4c51e169164bcc4187785dc1d560dc4). 

Let's explore this `richtext` > `inline_richtext` question separately in a followup PR. I'll mark [these todo decisions as completed for now](https://github.com/Shopify/dawn/pull/2085#issuecomment-1342851198), and also note this outcome in the decision log above. 

In the meantime, I'd welcome one more review to make sure nothing else seems off about this PR! Thanks. 🙌

**Review by ludoboludo — DISMISSED** [review]
I think it's looking good 👍

**Review by ludoboludo — DISMISSED** [review]
Ah yeah formatting conflict. We're now going to enable by default the prettier plugin to format our files on save FYI. So you might want to have it locally and enable it to format on save in the future 👍 
[Prettier](https://shopify.dev/themes/tools/liquid-prettier-plugin).

**Review by melissaperreault — COMMENTED** [review]
Thanks team for all the care and effort!

Some headings display the link color instead of displaying it only on hover
- [ ] [Rich text ](https://screenshot.click/15-18-27oy8-65eqz.png)
- [ ] [Multicolumn section](https://screenshot.click/15-21-nz2lv-pla1q.png) heading
- [ ] Blog posts section
- [ ] Collection list section

**Comment by kjellr** [comment]
Ok @melissaperreault + @ludoboludo, I pushed [a really small commit]([5e1989a](https://github.com/Shopify/dawn/pull/2085/commits/5e1989adcef17e20c03cb1544eaafd12960311ca)) to fix those issues. 

- For the Rich Text Heading, the issue was just specificity. I added [an additional rule](https://github.com/Shopify/dawn/blob/5e1989adcef17e20c03cb1544eaafd12960311ca/assets/base.css#L749) to ensure the `inline-richtext` color style overrides the standard rich text color when both classes are present. This seems reasonable to me — there's really no reason to assign both classes unless we wanted this behavior. 
- The other cases were all due to a conflicting link color assignment for the `title-wrapper-with-link` class. This class is only applied to four heading areas (Multicolumn, Collection List, Featured Blog, and Main Cart Items). Removing the color assignment fixes the issues for the three sections you called out, and has no effect on the Main Cart Items instance, because that one already includes an `underline-link` class on [the only link that's present](https://github.com/Shopify/dawn/blob/5e1989adcef17e20c03cb1544eaafd12960311ca/sections/main-cart-items.liquid#L29) within that heading. 

Should be ready for a re-review! 🙌

File: assets/base.css
Comment (kjellr) [reply assets/base.css:762]: That `banner--desktop-transparent` class (and these rules) are actually used for the Image Banner _and_ the Slideshow section when "Show container on desktop" is unchecked. I opted to include the rule once in `base.css` instead of duplicating these styles in both `section-image-banner.css` and `component-slideshow.css`.

Comment (ludoboludo) [reply assets/base.css:762]: Ah I see. The slideshow section is also using/loading `section-image-banner.css` at the moment. So I think it might still be a good idea to move the stylings to that file only 👍

Comment (kjellr) [reply assets/base.css]: Ah, good call, I missed that. Updated in d956d6f.

File: sections/rich-text.liquid
Comment (ludoboludo) [inline sections/rich-text.liquid:28]: I think we can go ahead and make this heading an `inline_richtext` instead or `richtext` here. 
We should be able to mitigate some potential loss of text for merchants: https://github.com/Shopify/shopify/pull/395536

And we should be good to go. We can loop in another dev to double check if we missed anything as well before merging.

Comment (kjellr) [reply sections/rich-text.liquid:28]: Since I believe this PR is all set aside from this single change, my inclination is that we should merge this, and then follow up with that change in a simple followup PR once https://github.com/Shopify/shopify/pull/395536 lands.

Comment (ludoboludo) [reply sections/rich-text.liquid:28]: Sounds good to me @kjellr 👍 Ill try and see if I can get a second dev to 👀

**Review by melissaperreault — DISMISSED** [review]
Thank you for the care and patience! 🚢

File: sections/rich-text.liquid
Comment (ludoboludo) [reply sections/rich-text.liquid:28]: Yeah let's wait for a second dev approval. I brought it up to @andrewetchen and he should be able to review. 
There were some questions about the performance impact in the editor apparently which we might able to confirm or deny with @stufreen though I'm not sure it would be a blocker anyway.

File: sections/image-banner.liquid
Comment (kjellr) [reply sections/image-banner.liquid:105]: Good catch! In this specific case, the text mimics `rte` styles intentionally for visual consistency reasons: to ensure this text shares styles with the supporting text on the Image & Text block. We considered changing this field to `richtext`, but avoided it for now [as outlined here](https://github.com/Shopify/dawn/pull/2085#issuecomment-1342851198) and [in the decision log](https://github.com/Shopify/dawn/pull/2085#issue-1431952952). We can always re-address in a followup PR if we change our minds. 👏

**Comment by kjellr** [comment]
I fixed a couple minor merge conflicts — now that we're post-Editions + Dawn 8.0's launch, let's take another look at this. 

@ludo, @stu, I think the only remaining concern here was a potential impact on render times if there are a lot of rich text fields on a template. Would you folks mind looking into that so we can close this out? Thank you! 🙌

**Comment by ludoboludo** [comment]
Alrighty, I've explored/tested render times and it didn't to be an issue. I tried with a little over 20 sections that had `text` headings and tested again with the same setup but with `inline_richtext`. 
Used an internal tool to check the `total time to render liquid` and numbers were: 
- `text` - 445ms
- `inline_richtext` - 365ms

I'll re review the PR either today or next week so we can finally get it into the theme 👍

**Comment by stufreen** [comment]
@ludoboludo @kjellr Yup I agree - after Ludo ran many tests with storefront renderer using a LOT of inline-richtext settings I don't see any significant performance impacts on rendering time. We can keep an eye on lighthouse scores and real user metrics but I really doubt we'll see any impact, especially since most pages and sections will be cached.

**Comment by kjellr** [comment]
Wonderful, thank you both!

**Review by ludoboludo — DISMISSED** [review]
:shipit:

**Comment by kjellr** [comment]
@ludoboludo any idea why the PR is failing tests all of a sudden? Not much has changed in the PR lately. 🤔

File: sections/multicolumn.liquid
Comment (stufreen) [inline sections/multicolumn.liquid:1]: Duplicate? Seems to already be added on line 3

File: sections/newsletter.liquid
Comment (stufreen) [inline sections/newsletter.liquid:2]: Also a duplicate?

**Review by stufreen — DISMISSED** [review]
LGTM with comments addressed

🎩  on os2-demo store on Chrome and looks great

File: sections/newsletter.liquid
Comment (kjellr) [reply sections/newsletter.liquid:2]: Good catch! Fixed this [and the other one](https://github.com/Shopify/dawn/pull/2085#discussion_r1114557785) in 7d9f077.

File: assets/base.css
Comment (kjellr) [reply assets/base.css]: That's true! Writing it this way was more concise, but it required an override and was perhaps more difficult to interpret overall. I've tidied this up 00292c9.

File: assets/section-image-banner.css
Comment (kjellr) [reply assets/section-image-banner.css:446]: It doesn't _technically_ need to. But in this case, we have multiple rules that are inherited correctly on desktop by default. 

We _could_ definitely remove the `max-width` media query here, but then we'd need to take corrective action on 750 and above to replicate the way it would've normally appeared. This doubled the CSS rules, and seemed heavy-handed to me (especially considering there are lots of other `(max-width: 749px)` breakpoints used in this file already). 

Basically, it's a choice between this: 

```
@media screen and (max-width: 749px) {
  .banner:not(.slideshow) .rte a,
  .banner:not(.slideshow) .inline-richtext a:hover,
  .banner:not(.slideshow) .rte a:hover {
    color: currentColor;
  }
}
```

... or this: 

```
.banner:not(.slideshow) .rte a,
.banner:not(.slideshow) .inline-richtext a:hover,
.banner:not(.slideshow) .rte a:hover {
  color: currentColor;
}

@media screen and (min-width: 750px) {
  .banner:not(.slideshow) .rte a,
  .banner:not(.slideshow) .rte a:hover {
    color: rgba(var(--color-link),var(--alpha-link));
  }

  .banner:not(.slideshow) .inline-richtext a:hover {
    color: var(--color-link);
  }
```

`inherit` doesn't work in this case, so you're stuck specifying the same variables that are assigned in the original rules from [`component-rte.css`](https://github.com/Shopify/dawn/blob/00292c97d067d1d160b4ac2e87d74d945fed26b9/assets/component-rte.css#L50) and [`base.css`](https://github.com/Shopify/dawn/blob/00292c97d067d1d160b4ac2e87d74d945fed26b9/assets/base.css#L753). All in all, I felt that just using the `max-width` rule was simpler. Happy to change if you feel strongly though.

File: sections/slideshow.liquid
Comment (ludoboludo) [inline sections/slideshow.liquid:2]: The footer is loading the `component-rte.css` asset. I think we should just remove it from anywhere else since the footer is always present anyway 🤷 
Otherwise, not a huge deal but this in the inspect tool is pretty annoying: 
<details>
<summary>Screenshot</summary>


</details>

**Review by ludoboludo — APPROVED** [review]
One comment but can be tackled as a follow up. I created a PR to change the rich text heading to a `inline_richtext` type. So it could be done/looked into there 👍

**Pull request merged**
